packages/smooth_app/lib/main.dart

```diff
@@ -100,6 +100,7 @@ Future<bool> _init1() async {
     daoString: DaoString(_localDatabase),
   );
   await callbackDispatcher(_localDatabase);
+  await UserManagementProvider().checkUserLoginValidity();
```

Comment:
You just cannot do that: call the server in the app init and `await`ing for a response. So if the connection is slow, the user just waits before seeing anything.
At least, please remove the `await`.

Reply:
Agreed, it was  wondering in my mind the rest of the day,